### PR TITLE
Java annotations

### PR DESCRIPTION
**Pull request by oxisto**
Basic implementation of Java annotations (not the declaration itself but the usage)

### CODE REVIEW

**Comment by oxisto** [comment]
Any objections on merging this?

File: src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CXXLanguageFrontend.java [hunk 1]
```diff
@@ -404,8 +415,8 @@ public void processAttributes(@NonNull Node node, @NonNull IASTAttributeOwner ow
     return list;
   }
 
-  private List<Expression> handleTokenList(IASTTokenList tokenList) {
-    List<Expression> list = new ArrayList<>();
+  private List<AnnotationMember> handleTokenList(IASTTokenList tokenList) {
+    List<AnnotationMember> list = new ArrayList<>();
 
     for (IASTToken token : tokenList.getTokens()) {
       if (token.getTokenType() == 6) {
```
Comment (konradweiss) [inline src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CXXLanguageFrontend.java:422]: Is there a constant for the token type that can be used for comparison or could we comment what the token type 6 means?

File: src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CXXLanguageFrontend.java [hunk 2]
```diff
@@ -418,22 +429,26 @@ public void processAttributes(@NonNull Node node, @NonNull IASTAttributeOwner ow
     return list;
   }
 
-  private Expression handleToken(IASTToken token) {
+  private AnnotationMember handleToken(IASTToken token) {
     String code = new String(token.getTokenCharImage());
 
+    Expression expression;
     switch (token.getTokenType()) {
       case 1:
```
Comment (konradweiss) [inline src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CXXLanguageFrontend.java:437]: Same here for the token type, constants would be better for readability or else we need comments

Comment (oxisto) [reply src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CXXLanguageFrontend.java:437]: No, there isn't.

File: src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CXXLanguageFrontend.java (hunk 1 above)
Comment (oxisto) [reply src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CXXLanguageFrontend.java:422]: No, there isn't.

Comment (oxisto) [reply src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CXXLanguageFrontend.java:423]: I have added comments

File: src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CXXLanguageFrontend.java (hunk 2 above)
Comment (oxisto) [reply src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CXXLanguageFrontend.java:438]: I have added comments

**Comment by konradweiss** [comment]
The rest looks fine to me, this only handles usage of Annotations right? Creation of Annotations through @interface and similar is not supported? We can merge this and create an Issue that handles that as well.

**Pull request merged**